File: programa1.py
#!/usr/bin/python3

def _main() -> None:
    enter = str(input(""))
    sum = 0
    while True:
        for i in enter:
            sum = int(i) + sum
        if sum >= 10:
            enter = str(sum)
            sum = 0
        else:
            break

    print(sum)

if __name__ == '__main__':
    _main()

File: programa1.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    string input;
    int sum = 0, number = 0;
    cin >> input;
    while(sum < 10) {
        for (int i = 0; i < input.length(); i++) {
            number = int(input[i]) - 48;
            sum += number;

        }
    }
        cout << sum;


    return 0;
}